SU2_CFD/src/driver_structure.cpp

```diff
@@ -441,7 +441,8 @@ CDriver::CDriver(char* confFile,
   if (rank == MASTER_NODE){
     ConvHist_file = new ofstream[nZone];
     for (iZone = 0; iZone < nZone; iZone++) {
-      output->SetConvHistory_Header(&ConvHist_file[iZone], config_container[ZONE_0], iZone);
+      output->SetConvHistory_Header(&ConvHist_file[iZone], config_container[iZone], iZone);
+      config_container[iZone]->SetHistFile(&ConvHist_file[iZone]);
```

Comment:
This is a first attempt, but in general terms I came to the realization that we need to completely redefine the output instead of doing hacks here and there. And that means redefining the output container - which should simplify also the config input for files. 

In particular, the history file definition is way too messy: there are so many checks, double checks and double (if (!if)) that it becomes impossible to add new options. Abstraction is the way to go - and for this we need a clear distinction between levels of the containers.